### PR TITLE
Add off road rate metric

### PR DESCRIPTION
**Pull request by freddyaboulton**
Implements OffRoadRate metric.

- For every prediction, computes the percentage of modes that are not completely contained by the drivable area. 
- On a previous PR, @holger-nutonomy commented that the eval code should use the global coordinates as opposed to the local coordinates. Since computing the OffRoadRate is simpler if the predictions are in global coordinates, I went ahead and changed the `compute_metrics.py`, and `physics_baselines.py` to output predictions in the global frame.
- Added unit tests

### CODE REVIEW

**Comment by freddyaboulton** [comment]
Didn't realize how big the json were. @holger-nutonomy, let me know what your thoughts are regarding adding git lfs to the repo. I can also mock the polygons and remove the json files.

**Comment by freddyaboulton** [comment]
@emwolff The OffRoadRate gives a score of 0.5 for the following example. Red trajectories go off the road.

off_road_example

File: python-sdk/nuscenes/eval/predict/metrics.py [hunk 1]
```diff
@@ -290,14 +298,54 @@ def shape(self):
...
 
-    def __call__(self, predictions, stacked_ground_truth, probabilities):
-        raise NotImplementedError("OffRoadRate not implemented!")
+    def __init__(self, helper: PredictHelper, aggregators: List[Aggregator]):
+        self._aggregators = aggregators
+        self.helper = helper
+        self.drivable_area_polygons = self.load_drivable_area_polygons(helper)
+
+    @staticmethod
+    def load_drivable_area_polygons(helper: PredictHelper) -> Dict[str, MultiPolygon]:
```
Comment (holger-nutonomy) [inline python-sdk/nuscenes/eval/predict/metrics.py]: No comments

File: python-sdk/nuscenes/eval/predict/metrics.py [hunk 2]
```diff
@@ -290,14 +298,54 @@ def shape(self):
...
+
+        maps: Dict[str, NuScenesMap] = load_all_maps(helper)
+
+        polygons = {}
+        for map_name, map_api in maps.items():
+            drivable_area_polygons = [map_api.extract_polygon(token) for record in map_api.drivable_area
+                                      for token in record['polygon_tokens']]
+
+            # To ensure the multipolygon is valid
+            polygons[map_name] = MultiPolygon(drivable_area_polygons).buffer(0)
```
Comment (holger-nutonomy) [inline python-sdk/nuscenes/eval/predict/metrics.py]: Maybe explain what buffer does.

File: python-sdk/nuscenes/eval/predict/tests/test_metrics.py [hunk 3]
```diff
@@ -1,8 +1,28 @@
...
+    del mock_helper
+
+    def load_polygon(filename: str):
+        return shape(json.load(open(os.path.join(DIRECTORY, filename)))).buffer(0)
+
+    DIRECTORY = os.path.dirname(os.path.realpath(__file__))
+    return {'singapore-onenorth': load_polygon('onenorth_drivable_area.json'),
+            'singapore-hollandvillage': load_polygon('holland_village_drivable_area.json'),
+            'singapore-queenstown': load_polygon('queenstown_drivable_area.json'),
+            'boston-seaport': load_polygon('boston_drivable_area.json')}
```
Comment (holger-nutonomy) [inline python-sdk/nuscenes/eval/predict/tests/test_metrics.py]: Where do these mock polygons come from? Can we just add the code to create them from the official map?

File: python-sdk/nuscenes/eval/predict/tests/test_metrics.py [hunk 4]
```diff
@@ -226,9 +254,80 @@ def test_DeserializeMetric(self):
...
+        predictions = np.array([[(1150.811356677105, 1598.0397224872172),
+                                (1158.783061670897, 1595.5210995059333),
+                                (1166.7543904812692, 1593.0012894706226),
+                                (1174.6895821186222, 1590.3704726754975)],
+                               [(1263.841977478558, 943.4546342496925),
+                                (1262.3235250519404, 944.6782247770625),
+                                (1260.8163412684773, 945.9156425437817),
+                                (1259.3272449205788, 947.1747683330505)]])
+        self._do_test('singapore-hollandvillage', predictions, 0)
+
```
Comment (holger-nutonomy) [inline python-sdk/nuscenes/eval/predict/tests/test_metrics.py]: Lots of space here

**Pull request merged**